packages/astro/src/content/vite-plugin-content-virtual-mod.ts

```diff
@@ -76,7 +76,7 @@ export function astroContentVirtualModPlugin({
 							hydratedComponents: [],
 							clientOnlyComponents: [],
 							scripts: [],
-							containsHead: true,
+							containsHead: false,
```

Comment:
This, it turns out, was also an important typo. `containsHead` will actually stop our propagation logic.